datafusion/functions/src/string/concat.rs

```diff
@@ -261,6 +262,23 @@ impl ScalarUDFImpl for ConcatFunc {
     fn documentation(&self) -> Option<&Documentation> {
         Some(get_concat_doc())
     }
+
+    fn output_ordering(&self, inputs: &[ExprProperties]) -> Result<SortProperties> {
+        Ok(inputs
+            .first()
+            .map(|first| first.sort_properties.clone())
+            .filter(|first_ordering| {
+                inputs
+                    .iter()
+                    .skip(1)
+                    .all(|input| input.sort_properties == *first_ordering)
+            })
+            .unwrap_or(SortProperties::Unordered))
+    }
+
+    fn output_preserves_lex_ordering(&self, inputs: &[ExprProperties]) -> Result<bool> {
```

Comment:
Why don't you directly return true here ?

Reply:
Resolved in 8949021.